sonic_installer/main.py

```diff
@@ -317,5 +331,92 @@ def cleanup():
     if image_removed == 0:
         click.echo("No image(s) to remove")
 
+# Upgrade docker image
+@cli.command()
+@click.option('-y', '--yes', is_flag=True, callback=abort_if_false,
+        expose_value=False, prompt='New docker image will be installed, continue?')
+@click.option('--cleanup_image', is_flag=True, help="Clean up old docker image")
+@click.option('--enforce_check', is_flag=True, help="Enforce pending task check for docker upgrade")
+@click.option('--tag', type=str, help="Tag for the new docker image")
+@click.argument('container_name', metavar='<container_name>', required=True,
+    type=click.Choice(["swss", "snmp", "lldp", "bgp", "pmon", "dhcp_relay", "telemetry", "teamd"]))
+@click.argument('url')
+def upgrade_docker(container_name, url, cleanup_image, enforce_check, tag):
+    """ Upgrade docker image from local binary or URL"""
+
+    # example image: docker-lldp-sv2:latest
+    cmd = "docker inspect --format '{{.Config.Image}}' " + container_name
+    proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, shell=True)
+    (out, err) = proc.communicate()
+    if proc.returncode != 0:
+        sys.exit(proc.returncode)
+    image_latest = out.rstrip()
+
+    # example image_name: docker-lldp-sv2
+    cmd = "echo " + image_latest + " | cut -d ':' -f 1"
+    proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, shell=True)
+    image_name = proc.stdout.read().rstrip()
+
+    DEFAULT_IMAGE_PATH = os.path.join("/tmp/", image_name)
+    if url.startswith('http://') or url.startswith('https://'):
+        click.echo('Downloading image...')
+        try:
+            urllib.urlretrieve(url, DEFAULT_IMAGE_PATH, reporthook)
+        except Exception, e:
+            click.echo("Download error", e)
+            return
+        image_path = DEFAULT_IMAGE_PATH
+    else:
+        image_path = os.path.join("./", url)
+
+    # make sure orchagent is in clean state if swss is to be upgraded
+    if container_name == "swss":
+        skipPendingTaskCheck = " -s"
+        if enforce_check:
+            skipPendingTaskCheck = ""
+
+        cmd = "docker exec -it " + container_name + " orchagent_restart_check" + skipPendingTaskCheck
+        proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, shell=True)
+        (out, err) = proc.communicate()
+        if proc.returncode != 0:
+            if enforce_check:
+                click.echo("Orchagent is not in clean state, check syslog and try later")
+                sys.exit(proc.returncode)
+            else:
+                click.echo("Orchagent is not in clean state, upgrading it anyway")
+        else:
+            click.echo("Orchagent is in clean state and frozen for warm upgrade")
+
+    run_command("systemctl stop %s" % container_name)
+    run_command("docker rm  %s " % container_name)
```

Comment:
Remove extra space between `rm` and  `%s`

Reply:
done.